addons/sale_mrp/tests/test_sale_mrp_procurement.py

```diff
@@ -363,3 +375,47 @@ def test_sale_mrp_avoid_multiple_pickings(self):
         so.action_confirm()
         self.assertEqual(len(so.picking_ids), 1)
         self.assertEqual(so.picking_ids.picking_type_id, warehouse.out_type_id)
+
+    def test_routes_on_sale_order_line(self):
+        """ Ensure that when multiple routes are available on a product and sale order line,
+        the system considers routes on the Sale Order Line (SOL) and respects their priority over product routes.
+
+        For example, if MTO + Buy is selected on sale order line(while the product has a Manufacture route),
+        a Purchase Order should be created instead of triggering Manufacture.
+        """
+        if self.env['ir.module.module']._get('purchase').state != 'installed':
```

Comment:
I wonder if you could skip the dependency to `purchase` in this test by forcing a delivery from stock in the SO by directly selecting the delivery route from the warehouse, rather than going through the MTO-Manufacture that would be set on the product. 🤔